homeassistant/components/axis.py

```diff
@@ -0,0 +1,347 @@
+"""
+Support for Axis devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/axis/
+"""
+
+import json
+import logging
+import os
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+
+from homeassistant.const import (ATTR_LOCATION, ATTR_TRIPPED,
+                                 CONF_HOST, CONF_INCLUDE, CONF_NAME,
+                                 CONF_PASSWORD, CONF_SCAN_INTERVAL,
+                                 CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers.entity import Entity
+from homeassistant.loader import get_component
+
+
+# REQUIREMENTS = ['axis==5']
+REQUIREMENTS = ['http://github.com/Kane610/axis'
+                '/archive/v5.zip'
+                '#axis==5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'axis'
+CONFIG_FILE = 'axis.conf'
+
+AXIS_DEVICES = {}
+
+EVENT_TYPES = ['motion', 'vmd3', 'pir', 'sound',
+               'daynight', 'tampering', 'input']
+
+PLATFORMS = ['camera']
+
+AXIS_INCLUDE = EVENT_TYPES + PLATFORMS
+
+AXIS_DEFAULT_NAME = 'Axis network device'
+AXIS_DEFAULT_HOST = '192.168.0.90'
+AXIS_DEFAULT_USERNAME = 'root'
```

Comment:
Yes. I will mention in the documentation that all functions works on a viewer level account on the device and that is the recommended way out of a security perspective.